src/AppBundle/Service/OrgService.php

```diff
@@ -257,20 +281,29 @@ private function upsertClientFromCsv(array $row, EntityDir\User $userOrgNamed)
             }
 
             $this->added['clients'][] = $client->getCaseNumber();
-            $this->em->persist($client);
+
         }
+        $this->log('Setting named deputy on client to deputy id:' . $userOrgNamed->getId());
         $client->setNamedDeputy($userOrgNamed);
 
         // Add client to named user (will be done later anyway)
-        $userOrgNamed->addClient($client);
+        $client->addUser($userOrgNamed);
 
         // Add client to all the team members of all teams the user belongs to
         // (duplicates are auto-skipped)
-        foreach ($userOrgNamed->getTeams() as $team) {
-            foreach ($team->getMembers() as $member) {
-                $member->addClient($client);
+
+        $teams = $userOrgNamed->getTeams();
+        $depCount = 0;
+        foreach ($teams as $team) {
+            $members = $team->getMembers();
+            foreach ($members as $member) {
+                $client->addUser($member);
+                $depCount++;
             }
         }
+        $this->log('Assigned ' . $depCount . ' additional deputies to client');
+
+        $this->em->persist($client);
```

Comment:
This persist is only needed when the object has been created with `new` in this request - all objects retrieved from the repo are being managed by Doctrine by default - which is what `persist` does. Probably not a performance drain but good to know.